### ISSUE TITLE
Unable to send logs to azure application insights when using logback appender for java web project

### ISSUE THREAD

**Issue by deepushri** [Microsoft/ApplicationInsights-Java#453]
I have a java web app (springboot app) where in I want to push the logs to the application insights in azure. I am using logback as my logging framework as it is natively supported by Springboot.
I followed the following blog for the integration of the applicationInsights for my web app.

https://github.com/AzureCAT-GSI/DevCamp/tree/master/HOL/java/06-appinsights

As per this blog I have created the following:

 1. created the application insights resource in azure
 2. created the filter in my java web app to capture the http requests
 3. created ApplicationInsights.xml
 4. created logback.xml which will have the AI appender provided by Azure.

here are the snippets from my web app

1) ApplicationInsight filter class is as follows:-

    @Configuration
    
    public class AppInsightsConfig {
    
        @Bean
        public String telemetryConfig() {
            String telemetryKey = System.getenv("APPLICATION_INSIGHTS_IKEY");
            if (telemetryKey != null) {
                TelemetryConfiguration.getActive().setInstrumentationKey(telemetryKey);
            }
            return telemetryKey;
        }
    
        @Bean
        @Order(1)
        public org.springframework.boot.web.servlet.FilterRegistrationBean aiFilterRegistration() {
            FilterRegistrationBean registration = new FilterRegistrationBean();
            registration.setFilter(new WebRequestTrackingFilter());
            registration.addUrlPatterns("/**");
            registration.setOrder(1);
            return registration;
        } 
    
        @Bean(name = "WebRequestTrackingFilter")
        public Filter WebRequestTrackingFilter() {
            return new WebRequestTrackingFilter();
        }	
    }

----------
2) ApplicationInsights.xml (src/main/resources folder)

    <?xml version="1.0" encoding="UTF-8"?>
    <ApplicationInsights xmlns="http://schemas.microsoft.com/ApplicationInsights/2013/Settings" schemaVersion="2014-05-30">
    
    
        
        
        <InstrumentationKey>my app key</InstrumentationKey>
        
        
        
        
        <TelemetryModules>
            <Add type="com.microsoft.applicationinsights.web.extensibility.modules.WebRequestTrackingTelemetryModule"/>
            <Add type="com.microsoft.applicationinsights.web.extensibility.modules.WebSessionTrackingTelemetryModule"/>
            <Add type="com.microsoft.applicationinsights.web.extensibility.modules.WebUserTrackingTelemetryModule"/>
        </TelemetryModules>
        
        
        
        
        <TelemetryInitializers>
            <Add   type="com.microsoft.applicationinsights.web.extensibility.initializers.WebOperationIdTelemetryInitializer"/>
            <Add type="com.microsoft.applicationinsights.web.extensibility.initializers.WebOperationNameTelemetryInitializer"/>
            <Add type="com.microsoft.applicationinsights.web.extensibility.initializers.WebSessionTelemetryInitializer"/>
            <Add type="com.microsoft.applicationinsights.web.extensibility.initializers.WebUserTelemetryInitializer"/>
            <Add type="com.microsoft.applicationinsights.web.extensibility.initializers.WebUserAgentTelemetryInitializer"/>
        
        </TelemetryInitializers>
        </ApplicationInsights>  
    
     

-------------
3) Logback.xml is as follows:-

```xml
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <appender name="aiAppender" 
        class="com.microsoft.applicationinsights.logback.ApplicationInsightsAppender">
    </appender>
    <root level="trace">
        <appender-ref ref="aiAppender" />
    </root>

    <logger name="org.springframework.web" level="INFO"/>
</configuration>
```

The issue is that I am not able to push the logs to the app insights. However I am able to get the telemetries in the app insights.

Note: I am using the latest versions of appinsights libraries (core and web)

**Comment by dhaval24** [comment]
@deepushri thanks for letting us know. We are aware of the logback issue and will try to repro and address it in upcoming releases.

**Comment by dhaval24** [comment]
@deepushri have you tried if this issue persist with the latest released version of sdk v2.0.0-BETA. I believe that it should have been resolved. Let me know so we can close this issue

**Comment by dhaval24** [comment]
@deepushri is there any update on this issue from your end. We believe we have fixed this issue in our latest beta sdk and is good to close. I am waiting to hear from you if there is still something wrong. Let me know.

**Comment by dhaval24** [comment]
@deepushri @kmalhotra2504 we believe this issue has been fixed in all our subsequent releases after 1.0.10. You may try out the latest stable SDK 2.0.1. I am closing this as of now and you can always reopen if the issue persists.
